### PR TITLE
Fix issue with service deregistration using namespace

### PR DESCRIPTION
**Pull request by mkam**
Updates Consul API to the latest version (v1.12.0) to get the `ServiceDeregisterOpts` method that was introduced in v1.9.0. This allows us to set the namespace query parameter on a deregister service request.

Overview of Changes:
- Upgraded the Consul version
- Included query parameters when determining intercept path for `HttpIntercept`
- Updated registration manager to use namespace query parameter on deregistration
- Added compatibility test for checking CTS service registration successful
- Added v1.12.0 and updated 1.11.0-beta1-> 1.11.5 for Consul compatibility versions
- Removed the node check that was breaking with the new version of the client. `ExposedPort` was added to checks in [v1.10.0](https://github.com/hashicorp/consul/releases/tag/v1.10.0), and Catalog service registration requests were failing against versions older than 1.10.0 with `400 (Request decode failed: json: unknown field "ExposedPort")`. Looking at the test case, the node check wasn't being validated anyway and was grouped with the node name update.

Ran with compatibility test enabled on a separate branch, results can be seen [here](https://app.circleci.com/pipelines/github/hashicorp/consul-terraform-sync?branch=deregister-ns-compat&filter=all).

Fixes https://github.com/hashicorp/consul-terraform-sync/issues/873

### CODE REVIEW

File: client/consul_test.go [hunk 1]
```diff
@@ -49,7 +49,7 @@ func Test_GetLicense_API_Failure(t *testing.T) {
 
 	for _, tc := range cases {
 		t.Run(tc.name, func(t *testing.T) {
-			path := "/v1/operator/license"
+			path := "/v1/operator/license?signed=1"
```
Comment (wilkermichael) [inline client/consul_test.go:52]: can also set `signed=true` which might be more explicit?

File: testutils/utils.go [hunk 2]
```diff
@@ -322,7 +322,12 @@ type HttpIntercept struct {
 
 func NewHttpClient(t *testing.T, intercepts []*HttpIntercept) *http.Client {
 	f := func(req *http.Request) *http.Response {
-		intercept, err := find(intercepts, req.URL.Path)
+		path := req.URL.Path
+		query := req.URL.RawQuery
```
Comment (wilkermichael) [inline testutils/utils.go:326]: ah nice 👍

File: e2e/compatibility/consul_test.go [hunk 3]
```diff
@@ -537,6 +577,24 @@ func registerCatalog(t *testing.T, entity *capi.CatalogRegistration, port int) {
...
+	conf.Address = fmt.Sprintf("localhost:%d", port)
+	client, err := capi.NewClient(conf)
+	require.NoError(t, err)
+
+	checks, err := client.Agent().Checks()
+	if err != nil {
+		return nil, err
+	}
+	c, ok := checks[checkID]
+	if !ok {
```
Comment (wilkermichael) [inline e2e/compatibility/consul_test.go:579]: I noticed that when using getCheck and getService, the caller requires no error. Should we just use the passed in *testing.T to require true?
```Go
require.True(t, ok)
```

**Review by wilkermichael — APPROVED** [review]

**Comment by wilkermichael** [comment]
@mkam great work, just some minor comments

File: e2e/compatibility/consul_test.go (hunk 3 above)
Comment (mkam) [reply e2e/compatibility/consul_test.go:579]: I was leaving it to the caller for the future if we need to verify the opposite, where the desired behavior is that a check or service doesn't exist.

But I probably should have done an error assertion right before this in the case where getting the agent checks failed! We should always expect to get a list of checks back.

File: client/consul_test.go (hunk 1 above)
Comment (wilkermichael) [reply client/consul_test.go:52]: ah, all good then, when Consul team sent me the `uri` they used `signed=true`

**Pull request merged**